hstream/src/HStream/Server/Core/Subscription.hs

```diff
@@ -353,7 +353,10 @@ doSubInit ServerContext{..} subId = do
       ldCkpReader <-
         S.newLDRsmCkpReader scLDClient readerName ckpStoreId 5000 maxReadLogs
                             (Just ldReaderBufferSize)
-      -- ckpReader is used to deliver the subscription records. If there are no records to be delivered, just wait
+      -- Ideally, if the subscription has no data to deliver, ldCkpReader should block on the read call. However, in the current implementation,
+      -- a subscription forcing deletion operation requires that the sendRecords loop should `not` be blocked, otherwise the forcing deletion
+      -- would be blocked because the delete precondition cannot be met. So set a 1s timeout for ldCkpReader
+      S.ckpReaderSetTimeout ldCkpReader 1000
```

Comment:
Maybe you can call `stopReading` to stop the reader while force deleting?

Reply:
I'm actually not sure my understanding of the whole forced deletion logic is comprehensive and correct, this pr is just to fix the integration test. Personally, I prefer to fix other issues in future